transform.go

```diff
@@ -103,11 +103,19 @@ func transformImage(m image.Image, opt Options) image.Image {
 
 	// never resize larger than the original image
 	if !opt.ScaleUp {
-		if w > imgW {
-			w = imgW
+		if w >= imgW {
+			if h == 0 {
```

Comment:
Given the refactor, maybe break this into two commits (but in this same pull request)... one that refactors out the `resizeParams` func and adds some test cases for what it currently supports, and then a second commit that adds this new logic as well as a few additional tests cases.